sky/utils/controller_utils.py

```diff
@@ -786,10 +802,11 @@ def maybe_translate_local_file_mounts_and_sync_up(task: 'task_lib.Task',
 
         new_storage_mounts[
             file_mount_remote_tmp_dir] = storage_lib.Storage.from_yaml_config({
-                'name': file_bucket_name,
+                'name': bucket_name,
```

Comment:
Currently this PR uploads _all_ contents of workdir and file_mounts to the root of the bucket. This limits us from reusing the same bucket across tasks. Workdir and file_mounts also get mixed in the same dir, causing correctness issues. 

I feel we need to able to reuse the same bucket across tasks. To do so, we may need to create directories in the bucket and upload contents there to have some isolation across tasks and workdir/file_mounts.

E.g., workdir for task should get uploaded to `s3://my-bucket/<task hash or id>/workdir`, file_mounts should go to `s3://my-bucket/<task hash or id>/file_mounts/<mount_path>` and so on.

Reply:
Note that we would also need to manage the lifecycle of these files we upload, since we cannot have this bucket grow unbounded. 

We may need to have a mechanism to delete files after the task completes.

Reply:
I was about to say that. Thank you for bringing it up.

Reply:
Add an interface `sub_dir` to support mapping to different folders. Need your review about this, @romilbhardwaj @Michaelvll. Thanks!

Reply:
> Note that we would also need to manage the lifecycle of these files we upload, since we cannot have this bucket grow unbounded.

Most cloud storage APIs don't support directory deletion. Furthermore, we can't simply delete the bucket after a job finishes, since multiple jobs may share the same bucket after this PR.

Two approaches are possible:
1) After job finished, trigger a clean up process to query the bucket's contents under a specific subdirectory and delete them.
2) Delete the intermediate bucket when its idle, since multiple jobs may share the same bucket, we'll need a mechanism for bucket management to check if bucket is intermediate and currently in use. we could also add auto delete feature in this way.

However implementing auto deletion requires furthur discussion and design, suggest we keep the current PR as is and address that in follow up tasks.

Reply:
BTW, I can't find the code that actually runs the tasks. I'd like to see if we have a cleanup process for them. Could you point me to the relevant code?

Reply:
Cleanup happens since `storage.persistent` is set to False. 
https://github.com/skypilot-org/skypilot/blob/6fda9fd5717cfb08afd5ad7e8cff0a8f95bf6200/sky/backends/cloud_vm_ray_backend.py#L3506-L3511

https://github.com/skypilot-org/skypilot/blob/6fda9fd5717cfb08afd5ad7e8cff0a8f95bf6200/sky/data/storage.py#L940-L982

Reply:
Will add today

Reply:
Deletion support now.